src/cargo/ops/tree/graph.rs

```diff
@@ -579,7 +562,11 @@ fn add_cli_features(
     for fv in to_add {
         match fv {
             FeatureValue::Feature(feature) => {
-                let index = add_feature(graph, feature, None, package_index, EdgeKind::Feature).1;
+                let feature_edge = Edge {
+                    kind: EdgeKind::Feature,
+                    node: package_index,
+                };
```

Comment:
Should this be moved out from the for loop, along with line 599?

Reply:
I considered it but the weird thing is that on 599 we also have a `dep_edge` and I worried about pulling the definition of `feature_edge` away from that context.

Reply:
That is true. Too far away. Thanks!